app.js

```diff
@@ -1,16 +1,192 @@
-var http = require('http');
-var port = process.env.PORT || 2345;
+"use strict";
+const http = require('http');
+const url = require('url');
+const fs = require('fs');
+const path = require('path');
+const fuzzysort = require('fuzzysort');
+const geohash = require('latlon-geohash');
+const config = require('./config');
+
+//overload-protection module
+const protectCfg = {
+  production: process.env.NODE_ENV === 'production', // if production is false, detailed error messages are exposed to the client
+  clientRetrySecs: 1, // Client-Retry header, in seconds (0 to disable) [default 1]
+  sampleInterval: 5, // sample rate, milliseconds [default 5]
+  maxEventLoopDelay: 42, // maximum detected delay between event loop ticks [default 42]
+  maxHeapUsedBytes: 0, // maximum heap used threshold (0 to disable) [default 0]
+  maxRssBytes: 0, // maximum rss size threshold (0 to disable) [default 0]
+  errorPropagationMode: false // dictate behavior: take over the response
+                              // or propagate an error to the framework [default false]
+};
+const protect = require('overload-protection')('http', protectCfg);
+
+const port = config.port;
+
+//function that tests if latitude and longitude is valid (takes input pre-converted to Number)
+const testLatLon = function(latitude/*Number!*/,longitude/*Number!*/) {
+  return (
+  // watch out! false >= 0 === true
+  (latitude === 0 || (latitude > 0 && latitude <= 90) || (latitude < 0 && latitude >= -90)) ||
+  (longitude === 0 || (longitude > 0 && longitude <= 180) || (longitude < 0 && longitude >= -180))
+  );
+};
+
+//function that builds two arrays of geohash neighbours (first with geohash, second with score)
+const neighboursFunc = function(lat,long) {
+  let mainGeohash = geohash.encode(lat, long, 3);
+  let neighboursObj = geohash.neighbours(mainGeohash);
+  let neighboursArr = Object.keys(neighboursObj).map(function(n) { return neighboursObj[n]; });
+  let neighboursScoreArr = [];
+  //deal with main geohash
+  neighboursArr.unshift(mainGeohash);
+  neighboursScoreArr[0] = 100;
+  //deal with first neighbours
+  for (let i=1;i<neighboursArr.length;i++){
```

Comment:
something I really hate are those stupid c style iterators. Now that in ES6 we have:

```
const array = Array(neighboursArr.length).fill(75);
array[0] = 100;
```